libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateEngine.cs

```diff
@@ -216,5 +219,46 @@ private string GetOsPath(string ambigiousPath)
                 return ambigiousPath.Replace("\\", "/");
             }
         }
+
+        /// <summary>
+        /// Parse lg content to LGTemplate list.
+        /// All the imports of the content will be also tracked down to parse templates.
+        /// </summary>
+        /// <param name="content">Text content contains lg templates.</param>
+        /// <param name="name">Text name.</param>
+        /// <param name="importResolver">resolver to resolve LG import id to template text.</param>
+        /// <returns>LGTemplate list of parsed lg content.</returns>
+        private IList<LGTemplate> ParseContent(string content, string name, ImportResolverDelegate importResolver)
+        {
+            var sources = new Dictionary<string, LGResource>();
+            LoopLGText(content, name, sources, importResolver);
+            var parsedTemplates = sources.SelectMany(s => s.Value.Templates).ToList();
+
+            // check if there are duplicated templates among parsed templates from both current content and its imports
+            var checker = new StaticChecker(parsedTemplates);
```

Comment:
I don't think we should call staticChecker at this place, the whole design to make sure static checker is decoupled from parsing. Here we are coupling it again.

If we want to do this, do check on the result of this function, instead of doing it here